File: backend/src/models/Temple.js
import mongoose from "mongoose";

const pujaTimingSchema = new mongoose.Schema(
  {
    name: String,
    nameML: String,
    timeIST: String,
    description: String
  },
  { _id: false }
);

const templeSchema = new mongoose.Schema({
  name: {
    en: { type: String, required: true },
    ml: { type: String, required: true },
    sa: { type: String, required: true }
  },
  deity: { type: mongoose.Schema.Types.ObjectId, ref: "Deity", required: true },
  shortDescription: String,
  fullDescription: String,
  significance: String,
  tradition: { type: String, default: "Kerala Tantric Agama" },
  location: {
    city: { type: String, default: "Karunagapally" },
    district: { type: String, default: "Kollam" },
    state: { type: String, default: "Kerala" },
    country: { type: String, default: "India" },
    coordinates: {
      lat: { type: Number, default: 9.0167 },
      lng: { type: Number, default: 76.5333 }
    }
  },
  images: [String],
  heroImage: String,
  timings: {
    pujas: [pujaTimingSchema]
  },
  liveStreamUrl: String,
  isActive: { type: Boolean, default: true },
  nriNote: String,
  panchangLocation: {
    lat: { type: Number, default: 9.0167 },
    lng: { type: Number, default: 76.5333 },
    timezone: { type: String, default: "Asia/Kolkata" }
  }
});

export const Temple = mongoose.model("Temple", templeSchema);

File: shared/src/commonMain/kotlin/com/divya/data/models/Temple.kt
package com.divya.data.models

import kotlinx.serialization.Serializable

@Serializable
data class Temple(
    val id: String,
    val name: LocalizedText,
    val deity: Deity? = null,
    val shortDescription: String,
    val fullDescription: String,
    val significance: String,
    val tradition: String,
    val location: TempleLocation,
    val images: List<String> = emptyList(),
    val heroImage: String? = null,
    val timings: TempleTimings,
    val liveStreamUrl: String? = null,
    val isActive: Boolean = true,
    val nriNote: String,
    val panchangLocation: PanchangLocation,
)

@Serializable
data class TempleLocation(
    val city: String,
    val district: String,
    val state: String,
    val country: String,
    val coordinates: Coordinates,
)

@Serializable
data class TempleTimings(
    val pujas: List<TemplePujaTiming> = emptyList(),
)

@Serializable
data class TemplePujaTiming(
    val name: String,
    val nameML: String,
    val timeIST: String,
    val description: String,
)

@Serializable
data class PanchangLocation(
    val lat: Double,
    val lng: Double,
    val timezone: String,
)

